drivers/i2s/i2s_nrfx.c

```diff
@@ -596,14 +622,8 @@ static int i2s_nrfx_write(const struct device *dev,
 		return -EIO;
 	}
 
-	if (size != drv_data->tx.cfg.block_size) {
```

Comment:
It should be checked now that `size <= drv_data->tx.cfg.block_size`.

Reply:
The error condition is actually `size > drv_data->tx.cfg.block_size || size < sizeof(uint32_t)`.